### PR TITLE
feat(cactus-plugin-ledger-connector-cdl): add new connector plugin

### PR DESCRIPTION
**Pull request by outSH**
- Add new plugin for connecting with CDL service.
- It supports as `cactus-plugin-ledger-connector-cdl-socketio` which is now depracated and will be removed shortly.
- There are still no automatic tests for this package, only manual script that user can run against CDL instance to see if everything is working.
- Fix minor formatting issue in cbdc example (issue from lint)

**Pull Request Requirements**
- [ ] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [ ] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [ ] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [ ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ ] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

File: packages/cactus-plugin-ledger-connector-cdl/src/main/typescript/cdl-gateway.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        url,
+        responseType: "json",
+        headers: {
+          "User-Agent": userAgent,
+          "Content-Type": "application/json;charset=UTF-8",
+          ...authHeaders,
+        },
+        params: queryParams,
+        data: dataPayload,
+      });
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-cdl/src/main/typescript/cdl-gateway.ts:116]: baseUrl is part of configuration, dynamic part of url is html sanitized to prevent any funky business when it comes from a user

File: packages/cactus-plugin-ledger-connector-cdl/src/main/typescript/cdl-gateway.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      label: this.className,
+    });
+
+    const agentOptions: https.AgentOptions = {};
+
+    if (skipCertCheck && typeof skipCertCheck === "boolean") {
+      this.log.info(
+        `Allowing self signed CDL API GW certificates (skipCertCheck=${skipCertCheck})`,
+      );
+      agentOptions.rejectUnauthorized = false;
```
[the remark this answers is not in the document]
Comment (outSH) [reply packages/cactus-plugin-ledger-connector-cdl/src/main/typescript/cdl-gateway.ts:57]: It's enabled by default, disabling is a configuration option for test environment

**Review by petermetz — CHANGES_REQUESTED** [review]
@outSH The bundle validation CI job is failing with this:

```sh
/home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
	dist/cactus-plugin-ledger-connector-cdl.node.umd.min.js
	undefined

/home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
	dist/cactus-plugin-ledger-connector-cdl.web.umd.min.js
	undefined

```

**Comment by outSH** [comment]
> @outSH The bundle validation CI job is failing with this:
> 
> ```shell
> /home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
> 	dist/cactus-plugin-ledger-connector-cdl.node.umd.min.js
> 	undefined
> 
> /home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
> 	dist/cactus-plugin-ledger-connector-cdl.web.umd.min.js
> 	undefined
> ```

@petermetz I didn't include these package names because they were not created in the build process. Same seem to apply to all other connector plugins, min bundle creation fails with bunch of errors :/ For now I've added missing package names to please the CI. Please check if bundling is failing on your machine as well or not, and if so, we can create an issue to fix that (and maybe add a script to check if packages were in fact created).

**Review by petermetz — APPROVED** [review]
> > @outSH The bundle validation CI job is failing with this:
> > ```shell
> > /home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
> > 	dist/cactus-plugin-ledger-connector-cdl.node.umd.min.js
> > 	undefined
> > 
> > /home/runner/work/cacti/cacti/packages/cactus-plugin-ledger-connector-cdl/package.json: 
> > 	dist/cactus-plugin-ledger-connector-cdl.web.umd.min.js
> > 	undefined
> > ```
> 
> @petermetz I didn't include these package names because they were not created in the build process. Same seem to apply to all other connector plugins, min bundle creation fails with bunch of errors :/ For now I've added missing package names to please the CI. Please check if bundling is failing on your machine as well or not, and if so, we can create an issue to fix that (and maybe add a script to check if packages were in fact created).

I rebased onto upstream/main, then yarn install, yarn configure, finally I modified the scripts in the package.json to use `prod` as the webpack env parameter (see screenshot) and then ran `yarn build:dev` and it worked.
So based on all this I think it is working on my machine. It could be because of some change in the meantime on the main branch.
With all that said, the way you fixed the custom checks is also 100% OK with me, e.g. you declare the min file names but you don't have the scripts defined for the build so the minified webpack build will be skipped anyway. Other packages use the same workaround until we hash out the problems with the minification (which could take a long time so I didn't want us to be slowed down by this kind of issue)

TLDR: The way it is now looks good to me, thank you for the updates!

image

**Review by izuru0 — APPROVED** [review]
LGTM
